cpu/nrf52/radio/nrf802154/nrf802154.c

```diff
@@ -535,6 +514,25 @@ void isr_radio(void)
     cortexm_isr_end();
 }
 
+void nrf802154_setup(nrf802154_t *dev)
+{
+    netdev_t *netdev = (netdev_t*) dev;
+    netdev_ieee802154_t *netdev_ieee802154 = (netdev_ieee802154_t*) dev;
+    nrf802154_dev = netdev_ieee802154;
+
+    netdev->driver = &nrf802154_netdev_driver;
+#ifdef MODULE_GNRC
+#ifdef MODULE_GNRC_SIXLOWPAN
+    netdev_ieee802154->proto = GNRC_NETTYPE_SIXLOWPAN;
+#else
+    netdev_ieee802154->proto = GNRC_NETTYPE_UNDEF;
+#endif
+#endif
```

Comment:
Why does the driver have to deal with this?

Reply:
It shouldn't actually. I just copied what was already there. But this should be indeed handled by the upper layer.

Reply:
`netdev/ieee802154.c` already sets those values, so I think we can just drop this here.
Or would you rather do that in a separate PR?

Reply:
I can do it here :)

Reply:
done!

Reply:
Please squash :wink: